https/tls_config.go

```diff
@@ -164,14 +162,21 @@ func ConfigToTLSConfig(c *TLSStruct) (*tls.Config, error) {
 	return cfg, nil
 }
 
-// Listen starts the server on the given address. If tlsConfigPath isn't empty the server connection will be started using TLS.
-func Listen(server *http.Server, tlsConfigPath string, logger log.Logger) error {
-	if tlsConfigPath == "" {
+// TLSConfig contains the configuration required to start an tls/basic auth server.
```

Comment:
I'd like this to be more clear that it's a path to a yaml file.

Do we need to get complex enough that this is a struct?

Maybe just something like this:

```
type TLSConfigFilePath string
```